pages/wallet.js

```diff
@@ -279,7 +279,11 @@ function InvoiceScanner ({ fieldName }) {
           return (
             <QrScanner
               onDecode={(result) => {
-                helpers.setValue(result.replace(/^lightning:/, '').toLowerCase())
+                if (result.startsWith('bitcoin:')) {
+                  helpers.setValue(result.split('lightning=')[1].split(/[&?]/)[0].toLowerCase())
```

Comment:
Welcome to the team! This is great but I'm a little concerned about the `split('lightning=')[1]` being undefined if someone scans an onchain-only QR.

Perhaps we could check and display a [toast](https://github.com/stackernews/stacker.news/blob/master/components/toast.js)?

Reply:
Good point.

Maybe another point: I wasn't able to answer the question looking at [the grammar in BIP-21](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki#abnf-grammar) and simpler syntax if we can assume that `bitcoin:` will be at the front of BIP-21 invoices.

Reply:
Good ppoint, didn't even think about non-lightning bip21 qrs

Reply:
simplified it a bit as you pointed out we don't really need to determine its a bip21 through `bitcoin:` also added a warning when not a unified qr or not of the format `lightning:` or `Labc123`